File: Thesis version/c++/PluginAdaptor/Command.cc
#include "Command.h"
#include "Headers.h"

namespace We3Graph
{
	int Command::__getClientID()
	{
		return clientID_;
	}

	void Command::__setClientID(int clientID)
	{
		clientID_ = clientID;
	}

	int Command::__getID()
	{
		return ID_;
	}

	void Command::__setID(int ID)
	{
		ID_ = ID;
	}

	Command::Command(string name, vector<string> * parameters)
	{
		name_ = name;
		parameters_ = parameters;

		for (int i = parameters_->size(); i < 5; i++)
			parameters_->push_back("");
	}

	string Command::GetName()
	{
		return name_;
	}

	vector<string> & Command::GetParameters()
	{
		return *parameters_;
	}
}

File: Current version/c#/PluginAdaptor/Command.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace We3Graph
{
    /// <summary>
    /// This class represent a command sent or received from the server.
    /// </summary>
    public class Command
    {
        private string name_;
        private List<string> parameters_;
        private int clientID_;
        private int ID_;

        internal int __getClientID()
        {
            return clientID_;
        }

        internal void __setClientID(int clientID)
        {
            this.clientID_ = clientID;
        }

        internal int __getID()
        {
            return ID_;
        }

        internal void __setID(int ID)
        {
            this.ID_ = ID;
        }

        internal Command(string name, List<string> parameters)
        {
            this.name_ = name;
            this.parameters_ = parameters;

            for (int i = parameters_.Count; i < 5; i++)
                parameters_.Add("");
        }

        public string GetName()
        {
            return name_;
        }

        public List<string> GetParameters()
        {
            return parameters_;
        }

    }
}
